src/wrapper.c

```diff
@@ -297,6 +297,33 @@ int cgroup_add_value_bool(struct cgroup_controller *controller,
 	return ret;
 }
 
+int cgroup_remove_value(struct cgroup_controller * const controller,
+			const char * const name)
+{
+	int i;
+
+	for (i = 0; i < controller->index; i++) {
+		if (strcmp(controller->values[i]->name, name) == 0) {
```

Comment:
A minor improvement would be to `free` the `multiline_value`, to avoid a memleak, something like
```       
        for (i = 0; i < controller->index; i++) {
                if (strcmp(controller->values[i]->name, name) == 0) {
+
+                       if (controller->values[i]->multiline_value)
+                               free(controller->values[i]->multiline_value);
+
                        if (i == (controller->index - 1)) {
```

Reply:
Excellent find!  You are spot on.

Reply:
In fact, the problem is even more insidious.  The `memmove()` just below this line is wrong.  It's supposed to move the array of pointers forward one slot in the array.  But it's actually moving the contents of one pointer (and potentially garbage after it) forward into the contents of the pointer that's been removed.  So it's a memory leak, potentially a security hole (by copying wrong data), and potentially a memory corruption by overwriting the memory after the value that's been removed.  Ugh!!

Reply:
But I have a fix for it... Coming soon :)